components/list/list-item-drag-handle.js

```diff
@@ -0,0 +1,196 @@
+
+import '../button/button-icon.js';
+import '../icons/icon.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { buttonStyles } from '../button/button-styles.js';
+import { getFirstFocusableDescendant } from '../../helpers/focus.js';
+
+const keyCodes = Object.freeze({
+	END: 35,
+	HOME: 36,
+	UP: 38,
+	DOWN: 40,
+	SPACE: 32,
+	ENTER: 13,
+	ESC: 27,
+	TAB: 9,
+	LEFT: 37,
+	RIGHT: 39,
+
+});
+
+export const dragActions = Object.freeze({
+	first: 'last',
+	end: 'first',
+	up: 'up',
+	down: 'down',
+	active: 'keyboard-active',
+	save: 'keyboard-deactivate-save',
+	cancel: 'keyboard-deactivate-cancel',
+	nextElement: 'next-element',
+	previousElement: 'previous-element'
+});
+
+class listItemDragHandle extends LitElement {
+
+	static get properties() {
+		return {
+			keyboardActive: { type: Boolean, attribute: 'keyboard-active', reflect: true },
+			text: { type: String }
+		};
+	}
+
+	static get styles() {
+		return [ buttonStyles, css`
+			:host {
+				align-items: center;
+				display: flex;
+				margin: 0.25rem;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.d2l-list-item-drag-handle-dragger-button,
+			.d2l-list-item-drag-handle-keyboard-button {
+				display: grid;
+				grid-auto-columns: auto;
+				grid-auto-rows: 1fr 1fr;
+				margin: 0;
+				min-height: 1.8rem;
+				padding: 0;
+				width: 0.9rem;
+			}
+			/* Firefox includes a hidden border which messes up button dimensions */
+			button::-moz-focus-inner {
+				border: 0;
+			}
+			.d2l-list-item-drag-handle-dragger-button {
+				background-color: unset;
+			}
+			button,
+			button[disabled]:hover,
+			button[disabled]:focus {
+				background-color: var(--d2l-color-gypsum);
+				color: var(--d2l-color-ferrite);
+			}
+			button:hover,
+			button:focus {
+				background-color: var(--d2l-color-mica);
+			}
+			button[disabled] {
+				cursor: default;
+				opacity: 0.5;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.keyboardActive = false;
+	}
+
+	focus() {
+		const node = getFirstFocusableDescendant(this);
+		if (node) {
+			node.focus();
+		}
+	}
+
+	render() {
+		return html`${this.keyboardActive ? this.renderKeyboardDragging() : this.renderDragger()}`;
+	}
+
+	renderDragger() {
+		return html`
+			<button class="d2l-list-item-drag-handle-dragger-button" @click="${this._handleDeactiveKeyboard}" @keydown="${this._handleDeactiveKeyboard}" aria-label="${this.text}">
+				<d2l-icon icon="tier1:dragger"></d2l-icon>
+			</button>
+		`;
+	}
+
+	renderKeyboardDragging() {
+		return html`
+			<button class="d2l-list-item-drag-handle-keyboard-button" @blur="${this._onBlur}" @keydown="${this._handleActiveKeyboard}" aria-label="${this.text}">
```

Comment:
@m6jones : Is there a reason we use `keydown` here instead of `keyup`?  Usually we prefer to use `keyup` for a11y.

@dlockhart : I know we went through and switched a bunch of our core components to use `keyup` instead since it's preferred.  Do you know the use-cases behind that preference? ... is it to allow cancelling the key somehow (and if so how do i, as a user, do it), is it something to do with modifier keys?

Reply:
It's not an accessibility thing AFAIK, but mouse down vs. up [definitely is](https://www.w3.org/WAI/standards-guidelines/wcag/new-in-21/#252-pointer-cancellation-a).

I think it's mostly that `keydown` isn't actually a "key press". Like if I keydown on something and don't release it for a second or two, when should the action occur -- when I first press down, or when I release? I'd expect on release, which would mean listening for `keyup`.

Reply:
I'll change it to keyup.

Reply:
You may or may not still need a `keydown` to `preventDefault`, depending on scrolling behavior (as per my other comment attached to that line).